packages/content/src/types.ts

```diff
@@ -0,0 +1,137 @@
+import { IItem, IGroup } from "@esri/arcgis-rest-types";
+
+export type HubType = 'document' | 'dataset' | 'map' | 'app' | 'site' | 'initiative' | 'event' | 'template';
+
+// Based on the [Hub Content UI Inventory](https://app.lucidchart.com/invitations/accept/dab82a25-2ed9-4f63-8075-3db8873087e3)
+// Attempting a simple _Hungarian Notation_ for attributes suffix: `*Date`, `*Id`, `*Number`. Plural (e.g. `tags`) is an array.
+
+export interface IContent extends IItem {
```

Comment:
what about events? they are not items.

Reply:
Technically, they are not items, but for purposes of this interface I'm proposing they look + act like items. 

I'm also OK if we say "no, Events, Members, Teams" are different and should have their own interfaces. but I think we want them to have common structure and attributes.

Reply:
I think the problem when you have a function that works on Item and then somebody sends in an event, that could be misleading and confusing.

Reply:
@tomwayson @mikeringrose _Events are a feature service_ is a logical implementation detail. Can you provide a concrete example of where the interface is fundamentally changed by this platform implementation? 

Events in particular may/should become items. but even before that the same domain model concepts apply (sharing, owner, etc.)

_arguing with myself_ if we conceptually say "Events are more like a Team than a Document" because Events have _members_ and _can attend_ actions then we could put events into the conceptual class of "Social Resources". Would that make more sense to an external developer and also our implementation?

Reply:
Yes. This is a library meant to be used in a larger ecosystem where an Item has a certain meeting and what you can do with an Item's id, such creating links to its resources. A person writing a method can _NOT_ make those assumptions if they expect the item may not really be an item.

Reply:
@mikeringrose I'm not clear how to interpret your message. 

Based on the concerns here, it's probably conceptually & technically simpler to keep `IHubEvent` independent of `IContent`. I'll pull it from this model definition.

Reply:
@ajturner i'm not sure what part you don't interpret, but you can leave it as is